devicetypes/smartthings/zwave-dual-metering-switch.src/zwave-dual-metering-switch.groovy

```diff
@@ -0,0 +1,262 @@
+	def value = cmd.value ? "on" : "off"
+	ep ? changeSwitch(ep, value) : []
+}
+
+private changeSwitch(endpoint, value) {
+	def result = []
+	if(endpoint == 1) {
+		result += createEvent(name: "switch", value: value, isStateChange: true, descriptionText: "Switch ${endpoint} is ${value}")
+	} else {
+		String childDni = "${device.deviceNetworkId}/$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+		child?.sendEvent(name: "switch", value: value, isStateChange: true, descriptionText: "Switch ${endpoint} is ${value}")
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd, ep) {
+	log.debug "Meter ${cmd}" + (ep ? " from endpoint $ep" : "")
+	def result
+	if(ep == 1) {
+		result = handleMeterReport(cmd)
+	} else if(ep) {
+		result = childHandleMeterReport(cmd, ep)
+	} else {
+		result = zwaveEvent(cmd)
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	def event = createEvent([isStateChange:  false, descriptionText: "Wattage change has been detected. Refreshing each endpoint"])
+	[event, response(refreshAll())]
+}
+
+def handleMeterReport(cmd) {
+	def result = []
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			result += createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kWh")
+			result += response(encap(zwave.meterV3.meterGet(scale: 2), 1))
+		} else if (cmd.scale == 2) {
+			result += createEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W")
+		}
+	}
+	result
+}
+
+def childHandleMeterReport(cmd, endpoint) {
+	String childDni = "${device.deviceNetworkId}/$endpoint"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	def result = []
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			child?.sendEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kWh")
+			result += response(encap(zwave.meterV3.meterGet(scale: 2), endpoint))
+		} else if (cmd.scale == 2) {
+			child?.sendEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W")
+		}
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd, ep) {
+	updateDataValue("MSR", String.format("%04X-%04X-%04X", cmd.manufacturerId, cmd.productTypeId, cmd.productId))
+	null
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, ep) {
+	log.warn "Unhandled ${cmd}" + (ep ? " from endpoint $ep" : "")
+}
+
+private onOffCmd(value, Integer endpoint = null) {
+	delayBetween([
+			encap(zwave.basicV1.basicSet(value: value), endpoint),
+			encap(zwave.basicV1.basicGet(), endpoint),
+			"delay 3000",
+			encap(zwave.meterV3.meterGet(scale: 2), endpoint)
+	])
+}
+
+private refreshCmd(endpoint) {
+	delayBetween([
+			encap(zwave.basicV1.basicGet(), endpoint),
+			encap(zwave.meterV3.meterGet(scale: 0), endpoint),
+			"delay 500"
+	], 500)
+}
+
+def on() {
+	onOffCmd(255, 1)
+}
+
+def off() {
+	onOffCmd(0, 1)
+}
+
+def refresh() {
+	def name = device.displayName.split(" ")
```

Comment:
this variable is unused

Reply:
Probably a leftover from testing and debugging, shouldn't have ended up here, sorry!